test/integration/clustercidr/ipam_test.go

```diff
@@ -134,6 +500,19 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
 			} else if !reflect.DeepEqual(test.expectedPodCIDRs, gotPodCIDRs) {
 				t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", test.expectedPodCIDRs, gotPodCIDRs)
 			}
+
+			// Delete the node.
+			if err := clientSet.CoreV1().Nodes().Delete(context.TODO(), test.node.Name, metav1.DeleteOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// Delete the Cluster CIDRs.
```

Comment:
should we do an active poll to verify this was really deleted, since we have to wait for the finalizer to be removed?

Reply:
I think we shoudl, or we can pollute the next test https://github.com/kubernetes/kubernetes/pull/115291/files#r1090862747

Reply:
Done, added poll to verify deletion of Node and ClusterCIDR objects